stateoftheuniverse/widgets/ExoplanetCount.py

```diff
@@ -0,0 +1,181 @@
+"""
+Get the current number of exoplanets (grouped by different categories).
+"""
+
+# -----------------------------------------------------------------------------
+# IMPORTS
+# -----------------------------------------------------------------------------
+
+import astropy.units as u
+
+from astroquery.nasa_exoplanet_archive import NasaExoplanetArchive
+from collections import Counter
+from datetime import datetime as dt
+from typing import Optional
+
+from stateoftheuniverse.widgets.prototypes import WidgetPrototype
+
+
+# -----------------------------------------------------------------------------
+# AUXILIARY FUNCTION DEFINITIONS
+# -----------------------------------------------------------------------------
+
+def get_exoplanet_class(mass: u.Quantity,
+                        radius: u.Quantity) -> str:
+    """
+    (Approximately) classify an exoplanet based on its mass and radius.
+
+    The classification scheme used here is based on the table from:
+        http://phl.upr.edu/library/notes/
+            amassclassificationforbothsolarandextrasolarplanets
+
+    Args:
+        mass: The mass of the exoplanet as a astropy.units.Quantity.
+        radius: The radius of the exoplanet as a astropy.units.Quantity.
+
+    Returns:
+        The approximate planet class, which is one of the following:
+            ["Asteroidan", "Mercurian", "Subterran", "Terran",
+             "Superterran", "Neptunian", "Jovian", "N/A", "Other"]
+    """
+
+    # Convert mass and radius to Earth units and cast to float
+    mass = mass.to(u.earthMass).to_value()
+    radius = radius.to(u.earthRad).to_value()
+
+    # Classify the planet based on its mass and radius
+    if (mass == 0) or (radius == 0):
+        return 'N/A'
+    elif (0 < mass <= 0.00001) and (0 < radius <= 0.03):
+        return 'Asteroidan'
+    elif (0.00001 <= mass <= 0.1) and (0.03 <= radius <= 0.7):
+        return 'Mercurian'
+    elif (0.1 <= mass <= 0.5) and (0.5 <= radius <= 1.2):
+        return 'Subterran'
+    elif (0.5 <= mass <= 2) and (0.8 <= radius <= 1.9):
+        return 'Terran'
+    elif (2 <= mass <= 10) and (1.3 <= radius <= 3.3):
+        return 'Superterran'
+    elif (10 <= mass <= 50) and (2.1 <= radius <= 5.7):
+        return 'Neptunian'
+    elif (50 <= mass < 5000) and (3.5 <= radius <= 27):
+        return 'Jovian'
+    else:
+        return 'Other'
+
+
+# -----------------------------------------------------------------------------
+# CLASS DEFINITIONS
+# -----------------------------------------------------------------------------
+
+class ExoplanetCount(WidgetPrototype):
+
+    def __init__(self,
```

Comment:
I think you don't have to define `__init__` here, it will be inherited from prototype. Or does it work differently for ABCs?

Reply:
If I try to remove the `__init__` method, this results in a
```
TypeError: Can't instantiate abstract class AstronomerBirthdays with abstract methods __init__
```
so maybe ABCs _are_ different after all? I'm definitely not an expert on this!